manifest/dynamic_manifest.go

```diff
@@ -31,169 +35,156 @@ type DynamicManifestProvider struct {
 	pubsub           *pubsub.PubSub
 	ec               ec.Backend
 	manifestServerID peer.ID
-	manifestTopic    *pubsub.Topic
 
-	// the lk guards all dynamic manifest-specific fields
-	lk               sync.RWMutex
-	manifest         Manifest
-	onManifestChange OnManifestChange
-	nextManifest     *Manifest
-}
+	runningCtx context.Context
+	errgrp     *errgroup.Group
+	cancel     context.CancelFunc
 
-func NewDynamicManifestProvider(manifest Manifest, pubsub *pubsub.PubSub, ec ec.Backend, manifestServerID peer.ID) ManifestProvider {
-	return &DynamicManifestProvider{
-		manifest:         manifest,
-		pubsub:           pubsub,
-		ec:               ec,
-		manifestServerID: manifestServerID,
-	}
+	initialManifest *Manifest
+	manifestChanges chan *Manifest
 }
 
-func (m *DynamicManifestProvider) Manifest() Manifest {
-	m.lk.RLock()
-	defer m.lk.RUnlock()
-	return m.manifest
+type ManifestUpdateMessage struct {
+	MessageSequence uint64
```

Comment:
I added a separate message sequence number.

1. Message sequence numbers order messages.
2. Versions only change when reconfiguring/rebootstrapping.

Reply:
Add comment on the Sequence vs Version